Marlin/Marlin_main.cpp

```diff
@@ -2962,6 +2990,23 @@ inline void gcode_M17() {
   enable_all_steppers();
 }
 
+#ifdef RESUME_FEATURE
+  inline void gcode_M19() {
+    if(code_seen('Z'))
+    {
+      gcode_get_destination(); // For Z
+      prepare_move();
```

Comment:
Actually… it occurs to me that Marlin updates the `current_position` to the end of the next planner move just after it sends it to the planner in `prepare_move`. Since `M114` reports that value, it will show the right numbers even without `st_synchronize`.